lib/data_magic.rb

```diff
@@ -113,10 +121,30 @@ def self.search(terms, options = {})
     end
 
     # assemble a simpler json document to return
+    simple_result =
     {
       "metadata" => metadata,
       "results" => 	results
     }
+
+    if options[:command] == 'stats'
```

Comment:
Yeah, I'm guessing each of the 3 `if` blocks could be a method, handling a specific aspect of the search request... but I thought the code was still on this side of becoming unwieldy within one method.